client/my-sites/themes/controller.jsx

```diff
@@ -77,19 +91,51 @@ export function loggedOut( context, next ) {
 	next();
 }
 
-export function fetchThemeData( context, next ) {
+export function fetchThemeData( context, next, shouldUseCache = false ) {
 	if ( ! context.isServerSide ) {
 		return next();
 	}
 
 	const queryParams = {
 		search: context.query.s,
 		tier: context.params.tier,
-		filter: context.params.filter,
+		filter: compact( [ context.params.filter, context.params.vertical ] ).join( ',' ),
 		page: 0,
 		perPage: PER_PAGE,
 	};
+	const cacheKey = context.path;
 
 	context.store.dispatch( query( queryParams ) );
-	context.store.dispatch( fetchNextPage( false ) ).then( () => next() );
+	context.store.dispatch( incrementThemesPage( false ) );
+
+	if ( shouldUseCache ) {
+		const cachedData = themesQueryCache.get( cacheKey );
+		if ( cachedData ) {
+			debug( `found theme data in cache key=${ cacheKey }` );
+			context.store.dispatch( cachedData.action );
+			context.renderCacheKey = context.path + cachedData.timestamp;
+			return next();
+		}
+	}
+
+	context.store.dispatch( fetchThemes( false ) )
```

Comment:
`fetchThemes( false )` doesn't read well to me, as I don't know what `false` represents until I look at the action. Could we just do `fetchThemes()`?